src/server/lib/callback-handler.js

```diff
@@ -203,22 +221,60 @@ export default async (sessionToken, profile, providerAccount, options) => {
             providerAccount.refreshToken,
             providerAccount.accessToken,
             providerAccount.accessTokenExpires
-          )
-          await dispatchEvent(events.linkAccount, { user, providerAccount })
+          );
+          await dispatchEvent(events.linkAccount, { user, providerAccount });
 
-          session = useJwtSession ? {} : await createSession(user)
-          isNewUser = true
+          session = useJwtSession ? {} : await createSession(user);
+          isNewUser = true;
           return {
             session,
             user,
-            isNewUser
+            isNewUser,
+          };
+        }
+      }
+    } else if (providerAccount.type === "webhook") {
+      // If signing in with an email, check if an account with the same email address exists already
+      const userByEmail = profile.phoneNumber
```

Comment:
`phoneNumber` is hardcoded here, need to find out a way to pass in the attribute to this method, and obviously names are off because of copying/pasting email behavior